repos/system_upgrade/common/actors/initramfs/upgradeinitramfsgenerator/libraries/upgradeinitramfsgenerator.py

```diff
@@ -21,6 +22,45 @@
 DRACUT_DIR = '/dracut'
 
 
+def _get_target_kernel_version(context):
+    """
+    Get the version of the most recent kernel version within the container.
+    """
+
+    kernel_version = None
+    try:
+        results = context.call(['rpm', '-qa', 'kernel'], split=True)
```

Comment:
Thanks! I did not know about that change. I did stumble upon the `InstalledTargetKernelVersion` model and wanted to make use of it, however, as it is produced in `RPMUpgradePhase`  the message is not produced when I need it. I was considering, whether it would be possible to move the actor backward in the workflow, but this is of course not possible, as the kernel version we need here needs to be queried from the context of the container, which is being created just now.

The way I chose to work around the issue, was basically emulate the way the kernel version was extracted in the [generate-initram.sh][1] script. But from what you are saying it seems that this might be broken as well?

So the only way forward I see right now is to either
 - copy-paste the way the kernel version is detected in `InstalledTargetKernelVersion` ,
 - generalize the version detection somewhere in a library, so that user is able to specify the context

and then to always pass it to the ``generate-initram.sh`` by a variable.

What would you suggest?

[1]: https://github.com/oamg/leapp-repository/blob/777e0a641739add1fca50af774d6d924af5550d7/repos/system_upgrade/common/actors/initramfs/upgradeinitramfsgenerator/files/generate-initram.sh#L9

Reply:
I've lied, you don't have to deal with `kernel-rt` at all, because regardless of whether the source system is real-time or not, the upgrade kernel is non-rt, the packages to install into the target userspace are specified [here](https://github.com/oamg/leapp-repository/blob/777e0a641739add1fca50af774d6d924af5550d7/repos/system_upgrade/common/actors/commonleappdracutmodules/libraries/modscan.py#L17). Therefore, there is no need to take care of kernel-rt, and it should be sufficient to just check for kernel-core. Truth be told, both `kernel` and `kernel-core` are being installed, so no change is required (although I would rather the query to be for `kernel-core` just to be sure).